src/gaps/GapsReportBuilder.ts

```diff
@@ -614,35 +614,42 @@ export function addFiltersToDataRequirement(
   withErrors: GracefulError[]
 ) {
   if (q.queryInfo) {
-    const detailedFilters = flattenFilters(q.queryInfo.filter);
-
-    detailedFilters.forEach(df => {
-      if (df.type === 'equals' || df.type === 'in') {
-        const cf = generateDetailedCodeFilter(df as EqualsFilter | InFilter, q.dataType);
-
-        if (cf !== null) {
-          dataRequirement.codeFilter?.push(cf);
-        }
-      } else if (df.type === 'during') {
-        const dateFilter = generateDetailedDateFilter(df as DuringFilter);
-        if (dataRequirement.dateFilter) {
-          dataRequirement.dateFilter.push(dateFilter);
-        } else {
-          dataRequirement.dateFilter = [dateFilter];
-        }
-      } else {
-        const valueFilter = generateDetailedValueFilter(df);
-        if (didEncounterDetailedValueFilterErrors(valueFilter)) {
-          withErrors.push(valueFilter);
-        } else if (valueFilter) {
-          if (dataRequirement.extension) {
-            dataRequirement.extension.push(valueFilter);
+    const relevantSource = q.queryInfo.sources.find(source => source.resourceType === q.dataType);
+    // if a source cannot be found that matches, exit the function
+    if (relevantSource) {
+      const detailedFilters = flattenFilters(q.queryInfo.filter);
+
+      detailedFilters.forEach(df => {
+        // DuringFilter, etc. inherit from attribute filter (and have alias on them)
+        if (relevantSource.alias === (df as AttributeFilter).alias) {
+          if (df.type === 'equals' || df.type === 'in') {
+            const cf = generateDetailedCodeFilter(df as EqualsFilter | InFilter, q.dataType);
+
+            if (cf !== null) {
+              dataRequirement.codeFilter?.push(cf);
+            }
+          } else if (df.type === 'during') {
+            const dateFilter = generateDetailedDateFilter(df as DuringFilter);
+            if (dataRequirement.dateFilter) {
+              dataRequirement.dateFilter.push(dateFilter);
+            } else {
+              dataRequirement.dateFilter = [dateFilter];
+            }
```

Comment:
Looks like in `DataRequirementHelpers.ts` this function gets called after calling `generateDataRequirement` on the retrieve. The [`generateDataRequirement` function](https://github.com/projecttacoma/fqm-execution/blob/89e1a89e5622cfb2d55d9413115e260f6fe26bc9/src/helpers/DataRequirementHelpers.ts#L332) already creates the `codeFilter` array if `retrieve.valueSet` or `retrieve.code` exists. So it seems that in most cases `dataRequirement.codeFilter` will already exist by the time that we get here, unless the retrieve does not have a code or valueSet. In that case, it seems that the `cf` here would not get added since `codeFilter` does not exist. Perhaps this is the expected behavior? @hossenlopp does this reasoning seem right?

Reply:
While this reasoning sounds good. It is possible that they may have not used a filter on the retrieve itself and may be filtering in the `where` part of the query. So `codeFilter` could definitely use similar logic to the `dateFilter` to see if the list is there first, and create it if it is not.